akka-stream-tests/src/test/resources/reference.conf

```diff
@@ -8,4 +8,5 @@ akka {
   akka.actor.warn-about-java-serializer-usage = false
 
   stream.materializer.debug.fuzzing-mode = on
+  stream.secret-test-fuzzing-warning-disable = 42
```

Comment:
OK. I just thought that this approach would be cleaner as it removes testing knobs from the API/settings. But the current warnings should work as well.

> Since these are the two places where ordinary people can learn about the existence of the feature...

A third place would be the API itself. ;)